hasura.planx.uk/migrations/1739222061613_update_view_applicant_research_opt_in/up.sql

```diff
@@ -0,0 +1,34 @@
+DROP VIEW IF EXISTS "public"."applicant_research_opt_in_view";
+CREATE OR REPLACE VIEW "public"."applicant_research_opt_in_view" AS 
+WITH data AS (
+    SELECT 
+        ls.id as planx_reference_number, 
+        ls.email, 
+        ls.created_at AS application_started_at, 
+        ls.submitted_at,
+        COALESCE(
+            ls.data->'passport'->'data'->'_address'->>'single_line_address',
+            ls.data->'passport'->'data'->'_address'->>'title'
+        ) as site_address,
```

Comment:
This `COALESCE` passport field logic is a good one to know about and commonly applies everywhere we display a site address (email templates, resume application, invite to pay etc - see methods like `getSessionDetails`, `buildContentFromSessions` in the API as examples using `||`).

If you look at the `MinimumSiteAddress` interface in the `FindProperty` component, you'll see that _every_ address will have a `title`, but this logic favors the `single_line_address` in cases of OS addresses because it's more detailed (eg includes town & postcode) and fallsback to `title` in cases where a new address is being proposed.